thamos/lib.py

```diff
@@ -433,6 +440,51 @@ def image_analysis(
     return response.result
 
 
+@with_api_client
+def build_analysis(
+    api_client: ApiClient,
+    build_detail: dict,
+    *,
+    environment_type: str,
+    registry_user: str = None,
+    registry_password: str = None,
+    registry_verify_tls: bool = True,
+    nowait: bool = False,
+    force: bool = False,
+    debug: bool = False,
+) -> typing.Union[typing.Dict, str]:
+    """Submit a build image and logs for analysis to Thoth."""
+    if not build_detail:
+        raise ValueError("No build info provided")
+
+    api_instance = BuildAnalysisApi(api_client)
+    if registry_user or registry_password:
+        # Swagger client handles None in a different way - we need to explicitly avoid passing
+        # registry user and registry password if they are not set.
+        response = api_instance.post_build(
+            body=build_detail,
```

Comment:
Can this be constructed explicitly? In other words - I, as a user of thamos.lib, I don't know what to pass to this function (`build_details` is too generic on this level). Can there be passed `log`, base image, output image explicitly as arguments and then, the request (which is a detail for thamos.lib consumer), is constructed in the python code of this function?

Reply:
@fridex  Do you mean having build_log:object, output_image:string and base_image:string as the input of this function and inside the function logic we pass them as the build_detail?

Reply:
Yes - the `build_detail` would be constructed internally inside the function.

Reply:
@fridex, the method is now updated

Reply:
@fridex the method is now updated